spec/services/create_project_activity_spec.rb

```diff
@@ -1,49 +1,96 @@
 require "rails_helper"
 
 RSpec.describe CreateProjectActivity do
-  let(:beis) { create(:beis_organisation) }
-  let(:delivery_partner_organisation) { create(:delivery_partner_organisation) }
-  let(:user) { create(:administrator, organisation: delivery_partner_organisation) }
-  let(:programme) { create(:programme_activity, organisation: beis) }
-
   describe "#call" do
-    let(:result) {
-      described_class.new(user: user, organisation_id: user.organisation.id, programme_id: programme.id).call
-    }
-
     it "sets the Organisation to that of users organisation" do
+      beis = create(:beis_organisation)
```

Comment:
I find this super interesting! This kind of repetition is what would usually have us reaching for `let` and here we think it is playing a part in making this fragile? 

Whilst I think I am fan of clear and simple over clever yet obfuscated, this is even making me feel odd!

I wonder two things:

- do we know that removing the mystery guests has fixed the issue?
- if it has, how do we prevent future us refactoring this test back to `lets` and returning this to it's fragile state?

I feel like there must be some way to setup a `create_project_activity` object with all the same attributes and then place it under test?

Reply:
You are quite right for flagging this. As this was an intermittent failing test that I noticed whilst trying to fix another problem I am going to defer the problem to a seperate task in order to get the cookie error fixed asap.